experimental/entity/kind.go

```diff
@@ -0,0 +1,80 @@
+package entity
+
+import (
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+)
+
+type KindRegistry interface {
+	Get(id string) Kind
+	GetFromSuffix(path string) Kind
+	List() []Kind
+	Register(kinds ...Kind) error
+}
+
+type KindInfo struct {
+	ID          string `json:"ID,omitempty"`
+	Description string `json:"description,omitempty"`
+	Category    string `json:"category,omitempty"`
+	// Detect the kind from a path suffix
+	PathSuffix string `json:"pathSuffix,omitempty"`
+
+	// For kinds with secure keys -- the keys will be strpped unless user has editor access
+	HasSecureKeys bool `json:"hasSecureKeys,omitempty"`
+
+	// The entity store does not extend the base EntityEnvelope -- this is typical for
+	// non-object-model formats like images (png, svg, etc)
+	IsRaw bool `json:"isRaw,omitempty"`
+
+	// For raw content types, this is set as an HTTP header
+	ContentType string `json:"contentType,omitempty"`
+}
+
+type Kind interface {
```

Comment:
A substantial portion of this is already expressed - and implemented - via `coremodel.Interface`. Why reimplement?

Reply:
this is just interface -- exception is that more robust systems (coremodels?) would implement

Why reimplement?  Really looking for the most general common patterns and while coremodels represents many things, it does not feel like a natural fit for binary data and things "typed" outside of cue -- or am I missing something?  images, svg, parquet, geojson, bluge segment files, csv are a few that come to mind immediatly